### PR TITLE
Custom Reports: Table Totals Callback

### PR DESCRIPTION
**Pull request by carkom**
Creating a callback for the table totals to fix a bug that created duplicate columns while rendering.

### CODE REVIEW

**Comment by Teprifer** [comment]
Hrm, I wonder if #2643 might be interfering here?

Using the test budget.

Starting from default:

image

Clicking on the total returns no results:

image

Change type to deposit:

image

Returns payments:

image

Change type to net:

image

Deposit total and payment total as above, total total returns deposit and payments.


---

Changing to Mode: Time

image

Jan - no results:

image

Change type to deposit - see payments and deposits:

image

Change type to net - see payments and deposits:

image

**Comment by carkom** [comment]
> Hrm, I wonder if #2643 might be interfering here?


Yea looks to be the case. Same issue with budget table? I might do a separate PR for it. This PR is concerned about the totals row call back and not to with the showActivity clicks.

**Comment by MatissJanis** [comment]
How can I reproduce the bug on edge?

**Comment by carkom** [comment]
Open a new custom report. Click table graph. Click time. Click a cell to go to accounts page. Click browser back button. Notice that the totals row has a bunch of 0s before the actual data starts.

File: packages/desktop-client/src/components/reports/graphs/tableGraph/ReportTableRow.tsx [hunk 1]
```diff
@@ -77,183 +86,195 @@ export const ReportTableRow = memo(
...
-            minWidth: compact ? 50 : 85,
-          }}
-          valueStyle={compactStyle}
-          width="flex"
-          privacyFilter
-        />
+        >
+          <Cell
+            value={item[groupByItem]}
+            title={item[groupByItem] || undefined}
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/reports/graphs/tableGraph/ReportTableRow.tsx:109]: **❓ question:** ‏is this necessary?
```suggestion
            title={item[groupByItem]}
```

**Review by MatissJanis — APPROVED** [review]

**Review by MatissJanis — APPROVED** [review]

**Pull request merged**